### ISSUE TITLE
[Bug]: Menu trigger does not have focus indicator if menu is opened by click and closed by Esc

### ISSUE THREAD

**Issue by jurokapsiar** [microsoft/fluentui#26124]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
https://react.fluentui.dev/?path=/docs/components-menu-menu--default
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://react.fluentui.dev/?path=/docs/components-menu-menu--default

### Bug Description

Click on menu trigger to open the menu
Press ESC to close the menu

## Actual Behavior
Focus is on trigger but the focus indicator is not visible

## Expected Behavior
Focus indicator should be visible as the focus was moved using keyboard

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
